PWGMM/Rivet/Tools/hepmc.py

```diff
@@ -0,0 +1,2642 @@
+        nvertices : str
+             Number of vertices
+        beam1 : str
+             ID of beam particle
+        beam2 : str
+             ID of beam particle
+        args : tuple
+             Remaning arguments.  Consist of
+
+                 [n_random [random]] [n_weights [weights]]
+
+        """
+        self._event = {
+            "number": int(number),
+            "nvertices": int(nvertices),
+            "nparticles": 0,
+            "attributes": {"mpi": mpi, "signal": [signal_id, signal_vertex]},
+            "alphaQCD": float(alpha_qcd),
+            "alphaQED": float(alpha_qed),
+            "event_scale": float(scale),
+            "vertices": {},
+            "particles": {},
+        }
+        nrandom = int(args[0])
+        if nrandom > 0:
+            self._event["attributes"]["random_states"] = [int(s) for s in args[1 : nrandom + 1]]  # X  # X
+
+        nweights = int(args[nrandom + 1])
+        if nweights > 0:
+            self._event["weighs"] = [float(w) for w in args[nrandom + 2 :]]
+
+    # ----------------------------------------------------------------
+    def _parse_info3(self, lineno, number, nvertices, nparticles, *args):
+        """Read header line with the format
+
+            event_number n_vertices n_particles [@ [position]
+
+        Parameters
+        ----------
+        lineno : int
+             Line number
+        number : str
+             Event number
+        nvertices : str
+             Number of vertices
+        nparticles : str
+             Number of particles
+        args : tuple of str
+             Additional arguments
+        """
+        self._event = {
+            "number": int(number),
+            "nvertices": int(nvertices),
+            "nparticles": int(nparticles),
+            "attributes": {},
+            "vertices": {},
+            "particles": {},
+        }
+
+        if len(args) <= 0:
+            return
+
+        if args[0] == "@":
+            self._event["shift"] = [float(f) for f in args[1:5]]
+
+    # ----------------------------------------------------------------
+    def _parse_units(self, lineno, energy, length, *args):
+        """Read units line with the format
+
+            energy length
+
+        Parameters
+        ----------
+        lineno : int
+             Line number
+        energy : str
+             Energy unit
+        length : str
+             Length unit
+        args : tuple of str
+             Additional arguments (should be empty)
+        """
+        self._assert_no_garbage(lineno, "units", args)
+
+        self._event["units"] = {"energy": energy, "length": length}  # pyright: ignore  # X  # X
+
+    # ----------------------------------------------------------------
+    def _make_vertex(
+        self,  # Do
+        vid=0,  # not
+        status=0,  # mess
+        position=None,  # with
+        incoming=None,  # my
+        outgoing=None,  # code
+        attributes=None,  # formatting
+        level=0,
```

Comment:
Yes there is.  Otherwise, the code formatter will put everything on one line making the code so much harder to read.  The comment asks people to back off.